core/src/G3Quat.cxx

```diff
@@ -430,10 +691,55 @@ pow(const G3TimestreamQuat &a, int b)
 }
 
 
+G3_SERIALIZABLE_CODE(Quat);
+G3_SERIALIZABLE_CODE(G3Quat);
 G3_SERIALIZABLE_CODE(G3VectorQuat);
 G3_SERIALIZABLE_CODE(G3TimestreamQuat);
+G3_SERIALIZABLE_CODE(G3MapQuat);
+G3_SERIALIZABLE_CODE(G3MapVectorQuat);
 
 namespace {
+static int
+Quat_getbuffer(PyObject *obj, Py_buffer *view, int flags)
+{
+	if (view == NULL) {
+		PyErr_SetString(PyExc_ValueError, "NULL view");
+		return -1;
+	}
+
+	view->shape = NULL;
+
+	bp::handle<> self(bp::borrowed(obj));
+	bp::object selfobj(self);
+	bp::extract<QuatPtr> ext(selfobj);
```

Comment:
Why use a shared pointer here instead of a reference (e.g. `Quat &`)? Using a shared pointer adds a bunch of overhead to no obvious purpose and requires defining the types QuatPtr etc.